src/vcpkg/vcpkgpaths.cpp

```diff
@@ -1161,17 +1166,16 @@ namespace vcpkg
             return format_filesystem_call_error(ec, "remove_all", {destination});
         }
 
-        Command git_archive = git_cmd_builder(dot_git_dir, git_tree_temp)
-                                  .string_arg("read-tree")
-                                  .string_arg("-m")
-                                  .string_arg("-u")
-                                  .string_arg(tree);
+        RedirectedProcessLaunchSettings git_archive{git_cmd_builder(dot_git_dir, git_tree_temp)
+                                                        .string_arg("read-tree")
+                                                        .string_arg("-m")
+                                                        .string_arg("-u")
+                                                        .string_arg(tree)};
 
-        auto env = default_environment;
+        auto& env = git_archive.environment.emplace();
```

Comment:
Is `default_environment` the same as a default constructed environment?

Reply:
https://github.com/microsoft/vcpkg-tool/blob/2f25fc8cd2e5a5444b1af2fa305ae8f725294ee4/src/vcpkg/base/system.process.cpp#L669-L670

Yes